---
int32 lower
int32 upper